src/core/stream/period/period_stream.ts

```diff
@@ -157,36 +165,46 @@ export default function PeriodStream({
       const newStream$ = clock$.pipe(
         take(1),
         mergeMap((tick) => {
+          const { audioTrackSwitchingMode } = options;
           const qSourceBuffer = createOrReuseQueuedSourceBuffer(sourceBuffersStore,
                                                                 bufferType,
                                                                 adaptation,
                                                                 options);
           const strategy = getAdaptationSwitchStrategy(qSourceBuffer,
                                                        period,
                                                        adaptation,
-                                                       tick);
+                                                       tick,
+                                                       audioTrackSwitchingMode);
           if (strategy.type === "needs-reload") {
             return observableOf(EVENTS.needsMediaSourceReload(period, tick));
           }
 
           const cleanBuffer$ = strategy.type === "clean-buffer" ?
             observableConcat(...strategy.value.map(({ start, end }) =>
                                qSourceBuffer.removeBuffer(start, end))
-                            ).pipe(ignoreElements()) :
-            EMPTY;
+                            ).pipe(last())
+                            : EMPTY;
 
           const bufferGarbageCollector$ = garbageCollectors.get(qSourceBuffer);
-          const adaptationStream$ = createAdaptationStream(adaptation, qSourceBuffer);
+          const adaptationStream$ = createAdaptationStream(adaptation,
+                                                           qSourceBuffer).pipe(share());
 
           return sourceBuffersStore.waitForUsableSourceBuffers().pipe(mergeMap(() => {
-            return observableConcat(cleanBuffer$,
+            return observableConcat(cleanBuffer$.pipe(
+              map(() =>
+               bufferType === "audio"
+                ? EVENTS.needsSourceBufferFlush(bufferType)
+                : EMPTY
+            )),
```

Comment:
Also we shake nowhere that the `audioTrackSwitchingMode` is set to `"flush"` so this just is the default behavior, no?

For me, the ideal thing to do would be:
  - `getAdaptationSwitchStrategy` tells us that we need to flush, by reading the type of adaptation, the buffer (you can take inspiration or even just copy from what we do when we decide to reload the video, I think) and `audioTrackSwitchingMode`, that we need to flush.

  - just based on what `getAdaptationSwitchStrategy` returns (that way we can easily adapt the behavior for another buffer type, just by updating that function her), we send that event.

Reply:
I see, u are right its cleaner! I committed with the new changes.

Reply:
For the comment on `last`, `concat` would be used here (so it would kind of be a `concat` with another `concat` inside, I let you the judge to make that readable)